### DISCUSSION TITLE
Вывод части текста отдельно regexp

### DISCUSSION THREAD

**Discussion by Makhmut2445** [totumonline/totum-ru-issues-and-discussions#201] [Как это сделать?] [answered]
Как можно это сделать? Эти строки собираются из других месте с помощью str и записываются сюда.
![image](https://user-images.githubusercontent.com/103171127/186859964-b5f5a5d5-e995-41f9-a5fc-6c0a4f4ba3ce.png)

**Comment by erpic-ru** [comment, marked as answer]
Странная задача должна иметь странное решение:

1. Если слово "Книга" идет всегда впереди, то следует подрезать строку через strPart на 5 символов
2. Затем разбить оставшуюся строку на элементы списка через [strSplit](https://ru.docs.totum.online/functions#fn-strSplit) по `"`
3. Взять первый элемент $split[0] и его уже разбить по точкам `.`
4. Взять первый элемент $split_2[0]

**Reply by ttm-alexeypolunin** [reply]
Я чуть-чуть отредактировал — это «лашпедос способ» конешн, но рабочий.

Будет проходить мимо нормальный программист я попрошу его сделать regexp, чтоб как у настоящих разработчиков было :)

**Reply by ttm-alexeypolunin** [reply]
Я бы еще strPart не первым делал, а последним и брал бы последние 4 символа, а не отрезал первые 5. 

Или если там разное количество цифр может быть или не всегда впереди 5 букв то вот так можно выбрать только цифры:

```
=: if(condition: $regm = true; then: $numbers) 

	regm: strRegAllMatches(str: $split_2; template: "[0-9]+"; matches: "matches"; flags: "u")

		numbers: listJoin(list: $#matches[0]; str: "")
		
		split_2: "book2015"
```

**Reply by Makhmut2445** [reply]
Нет, не всегда Книга стоит перед тем что нужно вывести.. Это может быть и по середине и в конце... Где угодно.

**Reply by ttm-alexeypolunin** [reply]
```
= : if(condition: $regm = true; then: $numbers) 

	regm: strRegAllMatches(str: $split_2[0]; template: "[0-9]+"; matches: "matches"; flags: "u")
		numbers: listJoin(list: $#matches[0]; str: "")
		
		split_2: strSplit(str: $split_1[$keynum]; separator: ".")
			keynum: $search[0]
				search: listSearch(list: $found; key: "value" = true)
					found: listReplace(list: $split_1; action: $found_book; value: "value")
						found_book: strRegAllMatches(str: $#value; template: "book")
						
		~split_1: strSplit(str: $string; separator: "|")
		
		string: "test data | book2015.ver_15 | log"
```
